### PR TITLE
feat(pci.storages.databases): datatr-44 add advanced roles mongoDB

### PR DESCRIPTION
**Pull request by ClaveauAlex**
| Question         | Answer
| ---------------- | ---
| Branch?          | master
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #DATATR-44
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

## Description

- Add new way to add roles for MongoDB

### :flags: Translations

- [x] TRANS-50188

### CODE REVIEW

File: packages/manager/modules/pci/src/projects/project/storages/databases/database/users/users.routing.js [hunk 1]
```diff
@@ -19,19 +20,21 @@ export default /* @ngInject */ ($stateProvider) => {
             database.engine,
             database.id,
           ).then((usersResponse) => map(usersResponse, (u) => new User(u))),
-        roles: /* @ngInject */ (DatabaseService, database, projectId) =>
-          DatabaseService.getRoles(
+        roles: /* @ngInject */ (DatabaseService, database, projectId) => {
+          return DatabaseService.getRoles(
```
Comment (anooparveti) [inline packages/manager/modules/pci/src/projects/project/storages/databases/database/users/users.routing.js:24]: **nitpick:** Unnecessary change

Comment (ClaveauAlex) [reply packages/manager/modules/pci/src/projects/project/storages/databases/database/users/users.routing.js]: Fixed :)

File: packages/manager/modules/pci/src/projects/project/storages/databases/database/users/add/add.controller.js [hunk 2]
```diff
@@ -29,12 +34,105 @@ export default class AddUserCtrl {
...
+  onAddAdvancedRole($index) {
+    const addedRole = this.model.selectedRoles[$index];
+    // check error
+    if (
+      !addedRole.admin &&
+      this.model.selectedRoles.filter(
+        (role) =>
+          role.key.name === addedRole.key.name &&
+          role.value === addedRole.value,
+      ).length > 1
```
Comment (anooparveti) [inline packages/manager/modules/pci/src/projects/project/storages/databases/database/users/add/add.controller.js:92]: **nitpick:** `Array.prototype.some` is more suitable since boolean is expected and the loop can be terminated after finding the first match.
```suggestion
      this.model.selectedRoles.some(
        (role) =>
          role.key.name === addedRole.key.name &&
          role.value === addedRole.value,
      )
```

File: packages/manager/modules/pci/src/projects/project/storages/databases/database/users/add/add.controller.js [hunk 3]
```diff
@@ -29,12 +34,105 @@ export default class AddUserCtrl {
...
+  static getRoles(roles) {
+    const newRolesList = [];
+    roles.map((role) =>
+      newRolesList.push({
+        name: role.name.split('@')[0],
+        admin: role.name.includes('admin'),
+        db: role.name.includes('admin') ? 'admin' : '',
+      }),
+    );
+    return newRolesList;
```
Comment (anooparveti) [inline packages/manager/modules/pci/src/projects/project/storages/databases/database/users/add/add.controller.js:124]:
```suggestion
    return roles.map((role) =>
      ({
        name: role.name.split('@')[0],
        admin: role.name.includes('admin'),
        db: role.name.includes('admin') ? 'admin' : '',
      }),
    );
```

File: packages/manager/modules/pci/src/projects/project/storages/databases/database/users/add/add.html [hunk 4]
```diff
@@ -64,8 +64,128 @@ <h2 data-translate="pci_databases_users_add"></h2>
...
+                                    <button
+                                        data-ng-disabled="!$ctrl.isButtonDeleteEnabled($index)"
+                                        type="button"
+                                        class="oui-inline-adder__action oui-button oui-button_primary oui-button_s"
+                                        data-ng-click="$ctrl.onRemoveAdvancedRole($index)"
+                                        data-ng-if="$ctrl.isButtonDeleteDisplayed($index)"
+                                    >
+                                        <span
+                                            class="oui-icon oui-icon-trash"
+                                        ></span>
```
Comment (anooparveti) [inline packages/manager/modules/pci/src/projects/project/storages/databases/database/users/add/add.html:167]:
```suggestion
                                        <span
                                            class="oui-icon oui-icon-trash"
                                            aria-hidden="true"
                                        ></span>
```

**Review by anooparveti — CHANGES_REQUESTED** [review]

File: packages/manager/modules/pci/src/projects/project/storages/databases/database/users/add/add.controller.js (hunk 3 above)
Comment (ClaveauAlex) [reply packages/manager/modules/pci/src/projects/project/storages/databases/database/users/add/add.controller.js]: Fixed 089fc4d282923269914a46ba902ecb1177dd7d41

File: packages/manager/modules/pci/src/projects/project/storages/databases/database/users/add/add.html (hunk 4 above)
Comment (ClaveauAlex) [reply packages/manager/modules/pci/src/projects/project/storages/databases/database/users/add/add.html]: Fixed 089fc4d282923269914a46ba902ecb1177dd7d41

File: packages/manager/modules/pci/src/projects/project/storages/databases/database/users/add/add.html [hunk 5]
```diff
@@ -64,8 +64,130 @@ <h2 data-translate="pci_databases_users_add"></h2>
 
     <div ng-if="!$ctrl.isRolesReadOnly">
         <h2 data-translate="pci_databases_users_add_roles"></h2>
+        <!--advanced roles-->
+        <div class="row" ng-if="$ctrl.isAdvancedRole">
```
Comment (jperchoc) [inline packages/manager/modules/pci/src/projects/project/storages/databases/database/users/add/add.html:68]: `data-ng-if`

File: packages/manager/modules/pci/src/projects/project/storages/databases/database/users/add/add.html [hunk 6]
```diff
@@ -64,8 +64,90 @@ <h2 data-translate="pci_databases_users_add"></h2>
...
+                                </footer>
+                            </div>
+                        </div>
+                    </div>
+                </form>
+            </div>
+        </div>
         <oui-dual-list
-            data-source="$ctrl.availableRoles"
+            ng-if="!$ctrl.isAdvancedRole"
```
[the remark this answers is not in the document]
Comment (ClaveauAlex) [reply packages/manager/modules/pci/src/projects/project/storages/databases/database/users/add/add.html]: Fixed b1cfdbcd0006505c24046a58b83d1929ae4ba2f1

File: packages/manager/modules/pci/src/projects/project/storages/databases/database/users/add/add.html (hunk 5 above)
Comment (ClaveauAlex) [reply packages/manager/modules/pci/src/projects/project/storages/databases/database/users/add/add.html]: Fixed b1cfdbcd0006505c24046a58b83d1929ae4ba2f1 :)

**Review by anooparveti — DISMISSED** [review]

**Review by jperchoc — APPROVED** [review]

**Review by jperchoc — APPROVED** [review]

**Review by anooparveti — DISMISSED** [review]

**Review by jperchoc — APPROVED** [review]

**Review by anooparveti — APPROVED** [review]
